Automattic/wp-calypso
# WPCOM: Migrate `wpcom.undocumented().site()` to `wpcom.req`

#### Changes proposed in this Pull Request

Currently, all  `wpcom.undocumented().site()` methods are used only from within the `requestSiteStats()` thunk. 

This PR migrates all the usages to `wpcom.req.get()` and removes the obsolete methods with their constructor and prototype.

Part of the ongoing effort to get rid of `wpcom.undocumented()`.

#### Testing instructions

* While testing the following, keep an eye on HTTP requests and verify all are still loading well and there are no 404s.
* Compare the following Calypso screens on this branch and trunk, and verify they work the same way a WP.com simple site, an Atomic site, and a site that has WooCommerce installed with some orders in it:
  * `/home/:site`
  * `/stats/day/:site`
  * other Stats pages - navigate to different Stats screens from the previous one.
  * `/store/stats/orders/day/:site` - only for a store with WooCommerce installed 
* Verify tests still pass: `yarn run test-client client/state/stats/lists/test/actions.js`
